api/src/main/kotlin/com/github/benmanes/gradle/versions/updates/resolutionstrategy/ComponentSelectionWithCurrent.kt

```diff
@@ -0,0 +1,19 @@
+package com.github.benmanes.gradle.versions.updates.resolutionstrategy
+
+import org.gradle.api.artifacts.ComponentSelection
+
+class ComponentSelectionWithCurrent @JvmOverloads constructor(
+  val currentVersion: String?,
+  private val delegate: ComponentSelection,
+) : ComponentSelection by delegate {
+
+  override fun toString(): String {
+    return """\
+ComponentSelectionWithCurrent{
+    group="${candidate.group}",
+    module="${candidate.module}",
+    version="${candidate.version}",
+    currentVersion="$currentVersion",
+}"""
```

Comment:
I think you can use `trimIndent()` so that it stays nicely formatted with the method's scoping, but drops the preceding whitespace.